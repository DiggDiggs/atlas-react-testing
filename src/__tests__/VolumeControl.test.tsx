import { describe, it, expect, vi } from "vitest";
import React from "react";
import { render } from "@testing-library/react";
import VolumeControl from "../components/VolumeControl";

// Mock the custom hook useVolumeControl
vi.mock("../hooks/useVolumeControl", () => ({
  __esModule: true,
  default: () => ({
    volume: 50,
    muteStatus: false,
    handleVolumeChange: vi.fn(),
    toggleMute: vi.fn(),
  }),
}));

describe("VolumeControl Component", () => {
  it("should match the snapshot when volume is not muted", () => {
    const { asFragment } = render(<VolumeControl />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should match the snapshot when volume is 0 (muted)", () => {
    // Mock the hook to simulate muted status
    vi.mock("../hooks/useVolumeControl", () => ({
      __esModule: true,
      default: () => ({
        volume: 0,
        muteStatus: true,
        handleVolumeChange: vi.fn(),
        toggleMute: vi.fn(),
      }),
    }));

    const { asFragment } = render(<VolumeControl />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should match the snapshot when volume is at max", () => {
    // Mock the hook to simulate volume at max
    vi.mock("../hooks/useVolumeControl", () => ({
      __esModule: true,
      default: () => ({
        volume: 100,
        muteStatus: false,
        handleVolumeChange: vi.fn(),
        toggleMute: vi.fn(),
      }),
    }));

    const { asFragment } = render(<VolumeControl />);
    expect(asFragment()).toMatchSnapshot();
  });
});
