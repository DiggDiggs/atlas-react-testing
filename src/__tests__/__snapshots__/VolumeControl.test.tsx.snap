// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControl Component > should match the snapshot when volume is 0 (muted) 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center"
  >
    <button
      class="volume"
    >
      <img
        alt="Volume Icon"
        class="size-7"
        src="/src/assets/volume.svg"
      />
    </button>
    <input
      class="h-8 w-full accent-steel-blue-700 outline-2 outline-offset-2 outline-transparent"
      id="volumeSlider"
      max="100"
      min="0"
      type="range"
      value="100"
    />
  </div>
</DocumentFragment>
`;

exports[`VolumeControl Component > should match the snapshot when volume is at max 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center"
  >
    <button
      class="volume"
    >
      <img
        alt="Volume Icon"
        class="size-7"
        src="/src/assets/volume.svg"
      />
    </button>
    <input
      class="h-8 w-full accent-steel-blue-700 outline-2 outline-offset-2 outline-transparent"
      id="volumeSlider"
      max="100"
      min="0"
      type="range"
      value="100"
    />
  </div>
</DocumentFragment>
`;

exports[`VolumeControl Component > should match the snapshot when volume is not muted 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center"
  >
    <button
      class="volume"
    >
      <img
        alt="Volume Icon"
        class="size-7"
        src="/src/assets/volume.svg"
      />
    </button>
    <input
      class="h-8 w-full accent-steel-blue-700 outline-2 outline-offset-2 outline-transparent"
      id="volumeSlider"
      max="100"
      min="0"
      type="range"
      value="100"
    />
  </div>
</DocumentFragment>
`;

exports[`VolumeControl Component should match the snapshot when volume is 0 (muted) 1`] = `
<div
  class="flex items-center justify-center"
>
  <button
    class="volume"
    type="button"
  >
    <img
      alt="Volume Icon"
      class="size-7"
      src="/path/to/mute.svg"
    />
  </button>
  <input
    class="h-8 w-full accent-steel-blue-700 outline-2 outline-offset-2 outline-transparent"
    id="volumeSlider"
    max="100"
    min="0"
    type="range"
    value="0"
  />
</div>
`;

exports[`VolumeControl Component should match the snapshot when volume is at max 1`] = `
<div
  class="flex items-center justify-center"
>
  <button
    class="volume"
    type="button"
  >
    <img
      alt="Volume Icon"
      class="size-7"
      src="/path/to/volume.svg"
    />
  </button>
  <input
    class="h-8 w-full accent-steel-blue-700 outline-2 outline-offset-2 outline-transparent"
    id="volumeSlider"
    max="100"
    min="0"
    type="range"
    value="100"
  />
</div>
`;

exports[`VolumeControl Component should match the snapshot when volume is not muted 1`] = `
<div
  class="flex items-center justify-center"
>
  <button
    class="volume"
    type="button"
  >
    <img
      alt="Volume Icon"
      class="size-7"
      src="/path/to/volume.svg"
    />
  </button>
  <input
    class="h-8 w-full accent-steel-blue-700 outline-2 outline-offset-2 outline-transparent"
    id="volumeSlider"
    max="100"
    min="0"
    type="range"
    value="50"
  />
</div>
`;
